website/blog/2023-02-21-ferretdb-compare-arrays.md

```diff
@@ -0,0 +1,380 @@
+
+This is also the case for the `$gt` operator which only selects the same BSON type.
+
+**Example:** Array Value Comparison
+
+To compare an array filter argument with array values, the elements of the array are iterated.
+At each index, the BSON comparison order is used to compare the values.
+
+:::note
+If the values at a particular index are not equal, that result is used as the final comparison result.
+If there are no more elements in the array during the iteration, then that array is considered smaller.
+:::
+
+To compare an array filter argument and an array field in a collection document, we compare each element in the arrays until we discover a difference.
+
+For example, the following steps show how to use the `$lt` filter to compare the filter argument `[3, 1, 'b']` with the array field `[3, 'b', 'a']` in a document.
+
+![Array comparison](/img/blog/array-comparison.jpg)
+
+1. Compare `[3, 1, 'b']` and `[3, 'b', 'a']` using comparison order.
+   They have the same BSON type; both are arrays.
+2. Compare the first index `3` and `3` using comparison order, both are numbers, and they are the same.
+3. Compare the next index, `1` and `'b'` using the BSON comparison order since they have different types, so `1` is a number and less than string `'b'`.
+
+So `[3, 1, 'b']` is less than `[3, 'b', 'a']`.
+
+:::note
+If an array has no more element to compare, the one with no more element is less.
+For the same reason, an empty array is less than arrays with elements.
+:::
+
+## Filter operator comparison
+
+In this section, we use `mongosh` to show examples of the comparison in filtering.
+
+### `$lt` operator comparison
+
+[$lt](https://docs.ferretdb.io/basic_operations/read/#retrieve-documents-using-operator-queries)
+filter operators select records that are less than the query.
+We use the following collection to demonstrate the `$lt` filter argument.
+
+```js
+db.ltexample.insertMany([
+  { _id: 'scalar-null', v: null },
+  { _id: 'scalar-number', v: 1 },
+  { _id: 'scalar-string', v: 'a' },
+  { _id: 'array-empty', v: [] },
+  { _id: 'array-null', v: [ null ] },
+  { _id: 'array-mixed', v: [ 1, 'z', 'b' ] },
+  { _id: 'array-string', v: [ 'd' ] },
+  { _id: 'array-bool', v: [ false ] },
+  { _id: 'nested-empty', v: [ [] ] },
+  { _id: 'nested-mixed', v: [ [ 'c' ], [ 1, 'z', 'b' ] ] },
+  { _id: 'nested-string', v: [ [ 'g' ] ] },
+  { _id: 'nested-bool', v: [ [ true ] ] }
+])
+```
+
+**Example:** `$lt` Operator with Scalar Argument
+
+When the filter argument is a string, documents with string records
+that has less value would be selected.
+
+```js
+db.ltexample.find({v: {$lt: 'c'}})
+```
+
+The output:
+
+```js
+[
+  { _id: 'scalar-string', v: 'a' },
+  { _id: 'array-mixed', v: [ 1, 'z', 'b' ] }
+]
+```
+
+The documents selected are `scalar-string` and `array-mixed`.
+
+The `scalar-string` document has a smaller value compared to the filter argument.
+
+The array record `array-mixed` was selected because the smallest element `b` is
```

Comment:
I think we need to bring the comparison between array and non array if want to use this example, we haven't explained how `array-mixed` got selected.

https://github.com/FerretDB/FerretDB/blob/bc761c53bb0ba74f2ce9e10870c6a223c6382c58/website/blog/2023-03-01.ferretdb-compare-arrays.md?plain=1#L123

Another option is remove everything array and nested array from the example.

Reply:
Thanks for pointing this out. I shouldn't have removed it. I'll bring it back now